sourced/ml/transformers/basic.py

```diff
@@ -153,13 +166,19 @@ def __setstate__(self, state):
         from bblfsh import Node
         self.parse_uast = Node.FromString
 
-    def __call__(self, rows: RDD) -> RDD:
+    def __call__(self, rows: Union[RDD, DataFrame]) -> RDD:
+        if isinstance(rows, DataFrame):
```

Comment:
Deserializer should always require an RDD. If it is a dataframe in your case, change the previous transfomer to return an rdd.